jdbc-sink/src/main/java/org/springframework/cloud/stream/module/jdbc/JdbcSinkConfiguration.java

```diff
@@ -0,0 +1,217 @@
+
+	@Autowired
+	private JdbcSinkProperties properties;
+
+	@Autowired
+	private JdbcMessageHandler msgHandler;
+
+
+	@Bean
+	public JdbcMessageHandler jdbcMessageHandler(DataSource dataSource) {
+		final MultiValueMap<String, Expression> columnExpressionVariations = new LinkedMultiValueMap<>();
+		for (Map.Entry<String, String> entry : properties.getColumns().entrySet()) {
+			String value = entry.getValue();
+			columnExpressionVariations.add(entry.getKey(), spelExpressionParser.parseExpression(value));
+			if (!value.startsWith("payload")) {
+				columnExpressionVariations.add(entry.getKey(), spelExpressionParser.parseExpression("payload." + value));
+			}
+		}
+		JdbcMessageHandler jdbcMessageHandler = new JdbcMessageHandler(dataSource,
+				generateSql(properties.getTableName(),columnExpressionVariations.keySet()));
+		jdbcMessageHandler.setSqlParameterSourceFactory(
+				new SqlParameterSourceFactory() {
+					@Override
+					public SqlParameterSource createParameterSource(Object o) {
+						if (!(o instanceof Message)) {
+							throw new IllegalArgumentException("Unable to handle type " + o.getClass().getName());
+						}
+						Message<?> message = (Message<?>) o;
+						MapSqlParameterSource parameterSource = new MapSqlParameterSource();
+						for (String key: columnExpressionVariations.keySet()) {
+							List<Expression> spels = columnExpressionVariations.get(key);
+							Object value = NOT_SET;
+							EvaluationException lastException = null;
+							for (Expression spel : spels) {
+								try {
+									value = spel.getValue(evaluationContext, message);
+								}
+								catch (EvaluationException e) {
+									lastException = e;
+								}
+							}
+							if (value == NOT_SET) {
+								if (lastException != null) {
+									logger.info("Could not find value for column '" + key + "': " + lastException.getMessage());
+								}
+								parameterSource.addValue(key, null);
+							}
+							else {
+								if (value instanceof JsonPropertyAccessor.ToStringFriendlyJsonNode) {
+									// Need to do some reflection until we have a getter for the Node
+									DirectFieldAccessor dfa = new DirectFieldAccessor(value);
+									JsonNode node = (JsonNode) dfa.getPropertyValue("node");
+									Object valueToUse;
+									if (node == null || node.isNull()) {
+										valueToUse = null;
+									}
+									else if (node.isNumber()) {
+										valueToUse = node.numberValue();
+									}
+									else if (node.isBoolean()) {
+										valueToUse = node.booleanValue();
+									}
+									else {
+										valueToUse = node.textValue();
+									}
+									parameterSource.addValue(key, valueToUse);
+								}
+								else {
+									parameterSource.addValue(key, value);
+								}
+							}
+						}
+						return parameterSource;
+					}
+				});
+		return jdbcMessageHandler;
+	}
+
+	@ServiceActivator(autoStartup = "false", inputChannel = Sink.INPUT)
+	public void handleMessage(Message<?> msg) {
```

Comment:
This isn't only redundant, but also some overhead.
The `@ServiceActivator` declaration should be moved to the `jdbcMessageHandler` `@Bean` definition.
The same concern which I raised in the Cassandra Sink PR before.